clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -4979,6 +4979,15 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
       CmdArgs.push_back("-fclangir-idiom-recognizer");
   }
 
+  if (Args.hasArg(options::OPT_fclangir_analysis_only_pipeline)) {
+    CmdArgs.push_back("-fclangir-analysis-only");
```

Comment:
What happens if both `-fclangir-analysis-only` and `-fclangir` are used in the same invocation? We need to clarify any assumption and possibly emit a diagnostic depending on the behavior we want. It's not clear to me yet what would be a good action here, ideas?

This makes me think that another possible solution is to add a `-fclangir-analysis` instead of `-fclangir-analysis-only`, which will run CIR analysis regardless if `-fclangir` is used. The tricky part here is that if `-fclangir` is used in conjunction, we probably want to reuse the generated CIR for analysis as part of LLVM lowering too, so we don't CIRGen twice.

Reply:
Yeah, I wrote this in the TODO comment. I'd like to split such part out of this patch to make it smaller. Currently, when both `-fclangir-analysis-only` and `-fclangir` are used, `-fclangir`  takes the higher priority. And I like to retain the behavior and emit an warning if both  `-fclangir-analysis-only` and `-fclangir` are detected.

I think when ClangIR gets more widely usable, there will be 3 pipelines:
- Traditional pipelines
- Traditional pipelines with clang IR analysis
- clangIR pipeline (with configurable passes)

In the clangIR pipeline, there are must be something to be checked by default. Otherwise it is really meaningless to the users. So I think `-fclangir-analysis-only` and `-fclangir` are mutual to each other. And we don't need to introduce `-fclangir-analysis`. Since the analysis should be by default enabled. Let's avoid adding too much driver options.

Reply:
Sounds good to me